cmd/xl-storage.go

```diff
@@ -853,49 +853,67 @@ func (s *xlStorage) ListDir(ctx context.Context, volume, dirPath string, count i
 	return entries, nil
 }
 
-func (s *xlStorage) deleteVersions(ctx context.Context, volume, path string, fis ...FileInfo) error {
+// StorageDeleteResp holds the response for storage delete operation
+type StorageDeleteResp struct {
+	Err  error
+	NoOp bool // true if delete marker already present and a soft delete is performed.
+}
```

Comment:
StorageDeleteResp should encapsulate the FileInfo of the delete marker that is the top most value so that we can avoid running getObjectInfo again on the disk.

Reply:
done